FirebaseAppDistribution/Sources/FIRAppDistribution.m

```diff
@@ -0,0 +1,291 @@
+                        dispatch_async(dispatch_get_main_queue(), ^{
+                          completion(nil, nil);
+                        });
+                      }
+                    }];
+
+  [listReleasesDataTask resume];
+}
+
+- (void)handleReleasesAPIResponseWithData:data
+                               completion:(FIRAppDistributionUpdateCheckCompletion)completion {
+  // TODO: Parse response from tester API, check instance identifier and maybe return a release
+
+  // Ensure we dispatch on the main thread to allow any UI to update
+  dispatch_async(dispatch_get_main_queue(), ^{
+    completion(nil, nil);
+  });
+}
+
+- (void)checkForUpdateWithCompletion:(FIRAppDistributionUpdateCheckCompletion)completion {
+  if (self.isTesterSignedIn) {
+    [self fetchReleases:completion];
+  } else {
+    UIAlertController *alert = [UIAlertController
+        alertControllerWithTitle:@"Enable in-app alerts"
+                         message:@"Sign in with your Firebase App Distribution Google account to "
+                                 @"turn on in-app alerts for new test releases."
+                  preferredStyle:UIAlertControllerStyleAlert];
+
+    UIAlertAction *yesButton =
+        [UIAlertAction actionWithTitle:@"Turn on"
+                                 style:UIAlertActionStyleDefault
+                               handler:^(UIAlertAction *action) {
+                                 [self signInTesterWithCompletion:^(NSError *_Nullable error) {
+                                   self.window.hidden = YES;
+                                   self.window = nil;
+
+                                   if (error) {
+                                     completion(nil, error);
+                                     return;
+                                   }
+
+                                   [self fetchReleases:completion];
+                                 }];
+                               }];
+
+    UIAlertAction *noButton = [UIAlertAction actionWithTitle:@"Not now"
+                                                       style:UIAlertActionStyleDefault
+                                                     handler:^(UIAlertAction *action) {
+                                                       // precaution to ensure window gets destroyed
+                                                       self.window.hidden = YES;
+                                                       self.window = nil;
+                                                       completion(nil, nil);
+                                                     }];
+
+    [alert addAction:noButton];
+    [alert addAction:yesButton];
+
+    // Create an empty window + viewController to host the Safari UI.
+    self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
+    self.window.rootViewController = [[UIViewController alloc] init];
+
+    // Place it at the highest level within the stack.
+    self.window.windowLevel = +CGFLOAT_MAX;
```

Comment:
This logic is defined twice. Can we have a function which takes in a UIViewController and it will handle updating the UIWindow?

Reply:
Done